core/cont/inc/TVirtualCollectionProxy.h

```diff
@@ -94,6 +94,14 @@ class TVirtualCollectionProxy {
       // Execute the container constructor
       return fClass.GetClass()==0 ? 0 : fClass->New(arena);
    }
+   virtual TClass::ObjectPtr NewObject() const {
+      // Return a new container object
+      return fClass.GetClass()==0 ? TClass::ObjectPtr{nullptr, nullptr} : fClass->NewObject();
```

Comment:
```suggestion
      return fClass.GetClass()==0 ? {} : fClass->NewObject();
```
works for me, too ;-)

Reply:
with gcc 8.4 and c++17 I get:
```
/home/pcanal/root_working/code/threaded-improvement/core/cont/inc/TVirtualCollectionProxy.h:120:36: error: expected ';' before '{' token
       return fClass.GetClass()==0 ? {} : fClass->NewObjectArray(nElements, arena);
                                    ^~
                                    ;
/home/pcanal/root_working/code/threaded-improvement/core/cont/inc/TVirtualCollectionProxy.h:120:40: error: expected primary-expression before ':' token
       return fClass.GetClass()==0 ? {} : fClass->NewObjectArray(nElements, arena);
                                        ^
```